tactics/tactics.ml

```diff
@@ -5038,6 +5038,18 @@ let tclABSTRACT ?(opaque=true) name_op tac =
     else name_op_to_name name_op (DefinitionBody Definition) "_subterm" in
   abstract_subproof ~opaque s gk tac
 
+let constr_eq x y =
+  Proofview.Goal.enter begin fun gl ->
+    let env = Tacmach.New.pf_env gl in
+    let evd = Tacmach.New.project gl in
+      match EConstr.eq_constr_universes env evd x y with
+      | Some csts ->
+        let csts = Universes.to_constraints ~force_weak:false (Evd.universes evd) csts in
+        if Evd.check_constraints evd csts then Proofview.tclUNIT ()
+        else Tacticals.New.tclFAIL 0 (str "Not equal")
```

Comment:
Perhaps this should get a different error message?  (e.g., "Not equal (universe mismatch)")